domain/src/main/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/domain/DomainAPI.kt

```diff
@@ -0,0 +1,56 @@
+package com.krzykrucz.elesson.currentlesson.preparedness.domain
+
+import com.krzykrucz.elesson.currentlesson.attendance.domain.CheckedAttendanceList
+import com.krzykrucz.elesson.currentlesson.attendance.domain.PresentStudent
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.UnpreparednessError.AlreadyRaised
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.UnpreparednessError.StudentNotPresent
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.StudentInSemesterReadError
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.StudentSubjectUnpreparednessInASemester
+import com.krzykrucz.elesson.currentlesson.shared.AsyncOutput
+import com.krzykrucz.elesson.currentlesson.shared.ClassName
+import com.krzykrucz.elesson.currentlesson.shared.CurrentLesson
+import com.krzykrucz.elesson.currentlesson.shared.FirstName
+import com.krzykrucz.elesson.currentlesson.shared.Output
+import com.krzykrucz.elesson.currentlesson.shared.SecondName
+
+data class UnpreparedStudent(
```

Comment:
according to `miro storming`, shouldn't ` UnpreparedStudent` be type of `Student` as well?

Reply:
it actually should be a type of PresentStudent

although, it would create some coupling, and this abstraction of a PresentStudent over the UnpreparedStudent type will probably not be used

what's more, it would require some rewriting in other places, because PresentStudent (which would become a sealed class) will be uninstantiable and would have to be replaced by PreparedStudent

Reply:
maybe I will put a TODO here

Reply:
avoiding coupling convince me :+1: